homeassistant/components/firmata/board.py

```diff
@@ -0,0 +1,131 @@
+"""Code to handle a Firmata board."""
+import logging
+from typing import Union
+
+from pymata_express.pymata_express import PymataExpress
+
+from homeassistant.const import CONF_NAME
+
+from .const import (
+    CONF_ARDUINO_INSTANCE_ID,
+    CONF_ARDUINO_WAIT,
+    CONF_BINARY_SENSORS,
+    CONF_SAMPLING_INTERVAL,
+    CONF_SERIAL_BAUD_RATE,
+    CONF_SERIAL_PORT,
+    CONF_SLEEP_TUNE,
+    CONF_SWITCHES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+FirmataPinType = Union[int, str]
+
+
+class FirmataBoard:
+    """Manages a single Firmata board."""
+
+    def __init__(self, config: dict):
+        """Initialize the board."""
+        self.config = config
+        self.api = None
+        self.firmware_version = None
+        self.protocol_version = None
+        self.name = self.config[CONF_NAME]
+        self.switches = []
+        self.binary_sensors = []
+        self.used_pins = []
+
+        if CONF_SWITCHES in self.config:
+            self.switches = self.config[CONF_SWITCHES]
+        if CONF_BINARY_SENSORS in self.config:
+            self.binary_sensors = self.config[CONF_BINARY_SENSORS]
+
+    async def async_setup(self, tries=0) -> bool:
+        """Set up a Firmata instance."""
+        try:
+            _LOGGER.debug("Connecting to Firmata %s", self.name)
+            self.api = await get_board(self.config)
+        except Exception as err:  # pylint: disable=broad-except
```

Comment:
Why did we change to catch `Exception` instead of `RuntimeError`?

Reply:
As I was testing how it fails, I found that `pymata-express` does not catch errors from the dependencies it uses, so it could through a `RuntimeError` or something not actually thrown by `pymata-express`.

Reply:
So what are those exceptions? `OSError`? `serial.SerialException`?

Reply:
I encountered `SerialException` but `SerialTimeoutException` would also be possible. I was unsure if it was bad practice to import from a dependency from a dependency. I'll add it in.